Tests/Functional/Controller/FrontEndEditorControllerTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TTN\Tea\Tests\Functional\Controller;
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\Test;
+use TTN\Tea\Controller\FrontEndEditorController;
+use TYPO3\TestingFramework\Core\Functional\Framework\Frontend\InternalRequest;
+use TYPO3\TestingFramework\Core\Functional\Framework\Frontend\InternalRequestContext;
+
+#[CoversClass(FrontEndEditorController::class)]
+final class FrontEndEditorControllerTest extends AbstractFrontendControllerTestCase
+{
+    private const PAGE_UID = 1;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->importCSVDataSet(__DIR__ . '/Fixtures/Database/FrontEndEditorController/ContentElement.csv');
+        $this->importCSVDataSet(__DIR__ . '/Fixtures/Database/FrontEndEditorController/FrontendUsers.csv');
+    }
+
+    #[Test]
+    public function indexActionForNoLoggedInUserRendersErrorMessage(): void
+    {
+        $request = (new InternalRequest())->withPageId(self::PAGE_UID);
+
+        $html = (string)$this->executeFrontendSubRequest($request)->getBody();
+
+        self::assertStringContainsString('Please configure this plugin to be only visible if a website user is logged in.', $html);
+    }
+
+    #[Test]
+    public function indexActionForLoggedInUserRendersTeaOwnedByTheLoggedInUser(): void
+    {
+        $this->importCSVDataSet(__DIR__ . '/Fixtures/Database/FrontEndEditorController/TeasAssignedToUser.csv');
+
+        $html = $this->getHtml();
+
+        self::assertStringContainsString('Godesberger Burgtee', $html);
+    }
+
+    #[Test]
+    public function indexActionForLoggedInUserDoesNotRenderTeaOwnedByOtherUser(): void
+    {
+        $this->importCSVDataSet(__DIR__ . '/Fixtures/Database/FrontEndEditorController/TeasAssignedToUser.csv');
+
+        $html = $this->getHtml();
+
+        self::assertStringNotContainsString('Oolong', $html);
+    }
+
+    #[Test]
+    public function indexActionForLoggedInUserDoesNotRenderTeaWithoutOwner(): void
+    {
+        $this->importCSVDataSet(__DIR__ . '/Fixtures/Database/FrontEndEditorController/TeasAssignedToUser.csv');
+
+        $html = $this->getHtml();
+
+        self::assertStringNotContainsString('Sencha', $html);
+    }
+
+    private function getHtml(): string
+    {
+        $request = (new InternalRequest())->withPageId(self::PAGE_UID);
+        $context = (new InternalRequestContext())->withFrontendUserId(1);
```

Comment:
Let's have a constant for the FE user UID.

Reply:
I'm not a fan to introduce another hop while reading code for a single location, which also already communicates the actual meaning of the hard coded value by the "word" in front of it `FrontendUserId`